### PR TITLE
Fix multiple validation with uploaded files

### PR DESCRIPTION
**Pull request by AdrienClairembault**
When submitting a validation request with an attached file to multiple users, only the first user will receive the file.

image

image

This is due to how we handle files: once an uploaded file is saved it is removed from the tmp file directory, so the first validation request will delete the file after using it and the others validation requests wont be able to find it.

To fix this, I've essentially duplicated the submitted files for each validation request so they can be treated independently from each others.

After fix, the document are handled as expected:

image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !29064

### CODE REVIEW

**Comment by AdrienClairembault** [comment]
Looking to validate that the proposed solution is acceptable before adding some testing (if possible).

File: src/Toolbox/DuplicatedFileUpload.php [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                // Replace prefix in file name
+                $old_prefix = $file['_prefix_filename'];
+                $old_filename = $file['_filename'];
+                $new_filename = $new_prefix . substr(
+                    $old_filename,
+                    strlen($old_prefix)
+                );
+
+                // Copy file on disk
+                copy(
```
Comment (cedric-anne) [inline src/Toolbox/DuplicatedFileUpload.php:174]: You should trigger a `E_USER_WARNING` if `copy` returns false.

**Review by cedric-anne — CHANGES_REQUESTED** [review]
As the `CronTask::cronTemp()` automatic action regularly cleans the temp directory, maybe it would be simplier to use `copy` instead of `rename` when a new documents is created from an uploaded file.

**Review by cedric-anne — APPROVED** [review]

**Comment by AdrienClairembault** [comment]
Impact will be one hour of uploaded documents in the tmp dir, could it negatively impact some installations with limited disk space ?

**Comment by cedric-anne** [comment]
> Impact will be one hour of uploaded documents in the tmp dir, could it negatively impact some installations with limited disk space ?

I do not think it will be a problem, unless there are many many many users that are uploading files very often. Administrator will be able to trigger the cron task every 5 minutes, or to limit uploaded documents size.

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**